components/camel-google-pubsub/src/main/java/org/apache/camel/component/google/pubsub/GooglePubsubEndpoint.java

```diff
@@ -0,0 +1,184 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.google.pubsub;
+
+import java.util.concurrent.ExecutorService;
+
+import com.google.api.client.repackaged.com.google.common.base.Strings;
+import com.google.api.services.pubsub.Pubsub;
+import org.apache.camel.Component;
+import org.apache.camel.Consumer;
+import org.apache.camel.ExchangePattern;
+import org.apache.camel.Processor;
+import org.apache.camel.Producer;
+import org.apache.camel.impl.DefaultEndpoint;
+import org.apache.camel.spi.Metadata;
+import org.apache.camel.spi.UriEndpoint;
+import org.apache.camel.spi.UriParam;
+import org.apache.camel.spi.UriPath;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * PubSub Endpoint Definition
+ */
+@UriEndpoint(scheme = "google-pubsub", title = "Google Pubsub",
+        syntax = "google-pubsub:projectId:destinationName?[options]", label = "messaging")
+public class GooglePubsubEndpoint extends DefaultEndpoint {
+
+    private Logger log;
+
+    @UriPath(description = "Project Id")
+    @Metadata(required = "true")
+    private String projectId;
+
+    @UriPath(description = "Destination Name")
+    @Metadata(required = "true")
+    private String destinationName;
+
+    @UriParam(name = "loggerId", description = "Logger ID to use when a match to the parent route required")
+    private String loggerId;
+
+    @UriParam(name = "concurrentConsumers", description = "The number of parallel streams consuming from the subscription")
```

Comment:
You need to set defaultValue in the `@UriParam` to 1 so its the same as the field value